tests/services/sleep_listener_test.py

```
@@ -0,0 +1,90 @@
+class NoQtImportTest(unittest.TestCase):
+    def test_module_imports_without_pyqt5(self):
+        # Import the module in a fresh subprocess with PyQt5 poisoned so any
+        # accidental Qt import would raise. Proves the headless path is Qt-free.
+        code = (
+            "import sys\n"
+            "sys.modules['PyQt5'] = None\n"
+            "import polyhost.services.sleep_listener as m\n"
+            "assert 'PyQt5' not in [k for k, v in sys.modules.items() if v is not None]\n"
+            "print('OK')\n"
+        )
+        proc = subprocess.run(
```

Ruff (0.15.15)

[error] `subprocess` call: check for execution of untrusted input

(S603)